src/journal-remote/journal-gatewayd.c

```diff
@@ -637,6 +753,63 @@ static int request_handler_fields(
         return MHD_queue_response(connection, MHD_HTTP_OK, response);
 }
 
+static int output_field_name(FILE *f, OutputMode m, const char *d) {
+        if (m == OUTPUT_JSON)
+                return fprintf(f, "{ \"field\" : \"%s\" }\n", d);
+        else
+                return fprintf(f, "%s\n", d);
+}
+
+static int request_handler_field_names(
+                struct MHD_Connection *connection,
+                void *connection_cls) {
+
+        _cleanup_(MHD_destroy_responsep) struct MHD_Response *response = NULL;
+        RequestMeta *m = connection_cls;
+        int r, fd;
+        FILE *tmp;
```

Comment:
```
_cleanup_close_ int fd = -1;
_cleanup_fclose_ FILE *tmp = NULL;
int r;
```

Reply:
It breaks the code: in the logs I get "microhttpd: Failed to send data in request for '/fields'.".

Note that there's a design decision behind this code : 
> Listing field names : I intially coded it with MHD_create_response_from_callback like in request_handler_entries and _fields, but it was copy-paste code (which is dirty) and **I thought that it was always quite a short list**, so I buffered all at once in a temporary file and output it with MHD_create_response_from_fd. Is it OK ? 

It's not always a short list, and may take time to execute. So maybe the first solution (copy-paste) is better, or maybe another level of abstraction : a function that takes one or several function as argument(s) and would be suitable for running `request_handler_entries`, `_fields` and `_field_names` (replacing `request_reader_entries` and `request_reader_fields`)